lib/auth/keystore/manager.go

```diff
@@ -326,14 +413,27 @@ func (m *Manager) GetJWTSigner(ctx context.Context, ca types.CertAuthority) (cry
 				return nil, trace.Wrap(err)
 			}
 			signer, err := backend.getSigner(ctx, keyPair.PrivateKey, pub)
-			return signer, trace.Wrap(err)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			return &cryptoCountSigner{Signer: signer, labelValue: labelJWTKey}, trace.Wrap(err)
 		}
 	}
 	return nil, trace.NotFound("no usable JWT key pairs found")
 }
 
 // NewSSHKeyPair generates a new SSH keypair in the keystore backend and returns it.
 func (m *Manager) NewSSHKeyPair(ctx context.Context, purpose cryptosuites.KeyPurpose) (*types.SSHKeyPair, error) {
+	createCounter.WithLabelValues(labelSSHKey).Inc()
+	key, err := m.newSSHKeyPair(ctx, purpose)
+	if err != nil {
+		createErrorCounter.WithLabelValues(labelSSHKey).Inc()
+		return nil, trace.Wrap(err)
+	}
+	return key, nil
+}
+
+func (m *Manager) newSSHKeyPair(ctx context.Context, purpose cryptosuites.KeyPurpose) (*types.SSHKeyPair, error) {
 	alg, err := cryptosuites.AlgorithmForKey(ctx, m.authPrefGetter, purpose)
```

Comment:
while we're here, it would be nice to track the key algorithm (RSA2048, ECDSAP256, or Ed25519) and the key type (software, pkcs11, AWS KMS, or GCP KMS, which you could find from backendForNewKeys)

Reply:
the `backend` type already has a `keyTypeDescription` but that's too verbose for a metric label, could add a similar method that just returns something like (software, pkcs11, aws_kms, or gcp_kms)

Reply:
key algorithm label added in [8c8b4b1](https://github.com/gravitational/teleport/pull/45338/commits/8c8b4b151a8a7d240145be5bd4c4764ac8dc5e56).

The counters are only incremented after the algorithm is known because I want the `create` counter to be a superset of `create_error`. We could consider a separate `key_algorithm_error` metric which would count any errors on the algorithm selection.

Let me know what you think.

Reply:
What you have looks good to me. algorithm selection should not be failing unless we have a serious bug